In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import re
import collections
from collections import defaultdict

# Loading the data

In [2]:
df_tweets = pd.read_csv("C:\\Users\\USER\\Documents\\Geetha\\Data Science\\Capstone Project 1\\tweets.csv", parse_dates=True)  

In [3]:
df_tweets = df_tweets.set_index(['time','name','username'])

In [4]:
df_tweets.head(20)


description  \
time            name          username                                                        
1/6/2015 21:07  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 21:27  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 21:29  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 21:37  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 21:45  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 21:51  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 22:04  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 22:06  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 22:17  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/10/2015 0:05  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/10/2015 0:08  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/12/2015 23:10 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/22/2015 20:14 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/22/2015 20:47 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/9/2015 10:54  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/12/2015 13:03 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/12/2015 13:11 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/16/2015 15:57 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/20/2015 16:28 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/20/2015 16:29 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   

                                              location  followers  \
time            name          username                              
1/6/2015 21:07  GunsandCoffee GunsandCoffee70      NaN        640   
1/6/2015 21:27  GunsandCoffee GunsandCoffee70      NaN        640   
1/6/2015 21:29  GunsandCoffee GunsandCoffee70      NaN        640   
1/6/2015 21:37  GunsandCoffee GunsandCoffee70      NaN        640   
1/6/2015 21:45  GunsandCoffee GunsandCoffee70      NaN        640   
1/6/2015 21:51  GunsandCoffee GunsandCoffee70      NaN        640   
1/6/2015 22:04  GunsandCoffee GunsandCoffee70      NaN        640   
1/6/2015 22:06  GunsandCoffee GunsandCoffee70      NaN        640   
1/6/2015 22:17  GunsandCoffee GunsandCoffee70      NaN        640   
1/10/2015 0:05  GunsandCoffee GunsandCoffee70      NaN        640   
1/10/2015 0:08  GunsandCoffee GunsandCoffee70      NaN        640   
1/12/2015 23:10 GunsandCoffee GunsandCoffee70      NaN        640   
1/22/2015 20:14 GunsandCoffee GunsandCoffee70      NaN        640   
1/22/2015 20:47 GunsandCoffee GunsandCoffee70      NaN        640   
2/9/2015 10:54  GunsandCoffee GunsandCoffee70      NaN        640   
2/12/2015 13:03 GunsandCoffee GunsandCoffee70      NaN        640   
2/12/2015 13:11 GunsandCoffee GunsandCoffee70      NaN        640   
2/16/2015 15:57 GunsandCoffee GunsandCoffee70      NaN        640   
2/20/2015 16:28 GunsandCoffee GunsandCoffee70      NaN        640   
2/20/2015 16:29 GunsandCoffee GunsandCoffee70      NaN        640   

                                               numberstatuses  \
time            name          username                          
1/6/2015 21:07  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:27  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:29  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:37  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:45  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:51 

In [5]:
df_tweets.info()


<class 'pandas.core.frame.DataFrame'>
MultiIndex: 17410 entries, (1/6/2015 21:07, GunsandCoffee, GunsandCoffee70) to (5/13/2016 16:10, one who believes, ro34th)
Data columns (total 5 columns):
description       14728 non-null object
location          11432 non-null object
followers         17410 non-null int64
numberstatuses    17410 non-null int64
tweets            17410 non-null object
dtypes: int64(2), object(3)
memory usage: 860.7+ KB


#### Columns 'description' and 'location' have null values

In [6]:
df_tweets['description'] = df_tweets['description'].fillna(' ')

In [7]:
df_tweets['location'] = df_tweets['location'].fillna(' ')

#### NaNs replaced with empty string in the columns 'description' and 'location' 

In [8]:
df_tweets.head(20)

description  \
time            name          username                                                        
1/6/2015 21:07  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 21:27  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 21:29  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 21:37  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 21:45  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 21:51  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 22:04  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 22:06  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/6/2015 22:17  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/10/2015 0:05  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/10/2015 0:08  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/12/2015 23:10 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/22/2015 20:14 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
1/22/2015 20:47 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/9/2015 10:54  GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/12/2015 13:03 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/12/2015 13:11 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/16/2015 15:57 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/20/2015 16:28 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   
2/20/2015 16:29 GunsandCoffee GunsandCoffee70  ENGLISH TRANSLATIONS: http://t.co/QLdJ0ftews   

                                              location  followers  \
time            name          username                              
1/6/2015 21:07  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 21:27  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 21:29  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 21:37  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 21:45  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 21:51  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 22:04  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 22:06  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 22:17  GunsandCoffee GunsandCoffee70                 640   
1/10/2015 0:05  GunsandCoffee GunsandCoffee70                 640   
1/10/2015 0:08  GunsandCoffee GunsandCoffee70                 640   
1/12/2015 23:10 GunsandCoffee GunsandCoffee70                 640   
1/22/2015 20:14 GunsandCoffee GunsandCoffee70                 640   
1/22/2015 20:47 GunsandCoffee GunsandCoffee70                 640   
2/9/2015 10:54  GunsandCoffee GunsandCoffee70                 640   
2/12/2015 13:03 GunsandCoffee GunsandCoffee70                 640   
2/12/2015 13:11 GunsandCoffee GunsandCoffee70                 640   
2/16/2015 15:57 GunsandCoffee GunsandCoffee70                 640   
2/20/2015 16:28 GunsandCoffee GunsandCoffee70                 640   
2/20/2015 16:29 GunsandCoffee GunsandCoffee70                 640   

                                               numberstatuses  \
time            name          username                          
1/6/2015 21:07  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:27  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:29  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:37  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:45  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:51 

In [9]:
df_tweets.info()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 17410 entries, (1/6/2015 21:07, GunsandCoffee, GunsandCoffee70) to (5/13/2016 16:10, one who believes, ro34th)
Data columns (total 5 columns):
description       17410 non-null object
location          17410 non-null object
followers         17410 non-null int64
numberstatuses    17410 non-null int64
tweets            17410 non-null object
dtypes: int64(2), object(3)
memory usage: 860.7+ KB


In [10]:
df_tweets=df_tweets.drop_duplicates()

In [11]:
df_tweets.info()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 17410 entries, (1/6/2015 21:07, GunsandCoffee, GunsandCoffee70) to (5/13/2016 16:10, one who believes, ro34th)
Data columns (total 5 columns):
description       17410 non-null object
location          17410 non-null object
followers         17410 non-null int64
numberstatuses    17410 non-null int64
tweets            17410 non-null object
dtypes: int64(2), object(3)
memory usage: 860.7+ KB


In [12]:
df_tweets.columns


Index(['description', 'location', 'followers', 'numberstatuses', 'tweets'], dtype='object')

In [13]:
df_tweets['description'] = df_tweets['description'].str.lower()
df_tweets['tweets'] = df_tweets['tweets'].str.lower()

In [14]:
df_tweets.head(20)


description  \
time            name          username                                                        
1/6/2015 21:07  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/6/2015 21:27  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/6/2015 21:29  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/6/2015 21:37  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/6/2015 21:45  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/6/2015 21:51  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/6/2015 22:04  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/6/2015 22:06  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/6/2015 22:17  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/10/2015 0:05  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/10/2015 0:08  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/12/2015 23:10 GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/22/2015 20:14 GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
1/22/2015 20:47 GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
2/9/2015 10:54  GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
2/12/2015 13:03 GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
2/12/2015 13:11 GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
2/16/2015 15:57 GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
2/20/2015 16:28 GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   
2/20/2015 16:29 GunsandCoffee GunsandCoffee70  english translations: http://t.co/qldj0ftews   

                                              location  followers  \
time            name          username                              
1/6/2015 21:07  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 21:27  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 21:29  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 21:37  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 21:45  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 21:51  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 22:04  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 22:06  GunsandCoffee GunsandCoffee70                 640   
1/6/2015 22:17  GunsandCoffee GunsandCoffee70                 640   
1/10/2015 0:05  GunsandCoffee GunsandCoffee70                 640   
1/10/2015 0:08  GunsandCoffee GunsandCoffee70                 640   
1/12/2015 23:10 GunsandCoffee GunsandCoffee70                 640   
1/22/2015 20:14 GunsandCoffee GunsandCoffee70                 640   
1/22/2015 20:47 GunsandCoffee GunsandCoffee70                 640   
2/9/2015 10:54  GunsandCoffee GunsandCoffee70                 640   
2/12/2015 13:03 GunsandCoffee GunsandCoffee70                 640   
2/12/2015 13:11 GunsandCoffee GunsandCoffee70                 640   
2/16/2015 15:57 GunsandCoffee GunsandCoffee70                 640   
2/20/2015 16:28 GunsandCoffee GunsandCoffee70                 640   
2/20/2015 16:29 GunsandCoffee GunsandCoffee70                 640   

                                               numberstatuses  \
time            name          username                          
1/6/2015 21:07  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:27  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:29  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:37  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:45  GunsandCoffee GunsandCoffee70              49   
1/6/2015 21:51 

In [15]:
df_tweets.index.get_level_values(1).value_counts(normalize=True)



Rami                    0.084721
War BreakingNews        0.068409
Conflict Reporter       0.062895
Salahuddin Ayubi        0.060655
Ibni Haneefah           0.040724
                          ...   
Khilafah Superstate     0.000402
Ibn Al-Shishani         0.000287
Abdul05                 0.000287
Newerajihadi61          0.000230
fahad slyman alsomal    0.000230
Name: name, Length: 112, dtype: float64

#### We are analyzing the tweets of 112 different users who have generated 17410 tweets

In [16]:

df_tweets['tweets'] =  df_tweets['tweets'] .apply(lambda elem: re.sub(r"(!|@|#|\n|')", "", elem)) 

#### Removing some of the special characters from 'tweets'column using regular expression operations

In [17]:
df_tweets['tweets'][6]



'english transcript : oh murabit : http://t.co/hujlj9kgkg http://t.co/t9ixmtbvgk'

In [18]:
df_tweets['tweets']

time             name              username       
1/6/2015 21:07   GunsandCoffee     GunsandCoffee70    english translation: a message to the truthful...
1/6/2015 21:27   GunsandCoffee     GunsandCoffee70    english translation: sheikh fatih al jawlani f...
1/6/2015 21:29   GunsandCoffee     GunsandCoffee70    english translation: first audio meeting with ...
1/6/2015 21:37   GunsandCoffee     GunsandCoffee70    english translation: sheikh nasir al wuhayshi ...
1/6/2015 21:45   GunsandCoffee     GunsandCoffee70    english translation: aqap: response to sheikh ...
                                                                            ...                        
5/13/2016 15:46  Anaksabil97       nvor85j            breakingislamicstate amaqagency fridaythe13th ...
5/13/2016 15:50  Anaksabil97       nvor85j            breaking iraqi forces bulldozer destroyed by g...
5/13/2016 15:51  Anaksabil97       nvor85j            breaking amaqagency islamicstate eastofhaditha...
5/13/2016 16:

In [19]:
'''

def df_tweets_tokenize(x):
    return x.str.split(' ').sum()
tweets_split= df_tweets.apply(df_tweets_tokenize(df_tweets['tweets']))   
'''

"\n\ndef df_tweets_tokenize(x):\n    return x.str.split(' ').sum()\ntweets_split= df_tweets.apply(df_tweets_tokenize(df_tweets['tweets']))   \n"

#### The tweets in 'tweets' column are splitted and are represened as a list in nw column 'tweets_split'

In [20]:
tweets_split= df_tweets['tweets'].str.split(' ').sum()
#print(tweets_split)
count_stopwords = collections.Counter(tweets_split)
#print(count_stopwords)

In [21]:

stopwords_morethan50 = dict()
for (key, value) in count_stopwords.items():
   # Check if key is even then add pair to new dictionary
   if value >= 50:
       stopwords_morethan50[key] = value
 

#print(stopwords_morethan50)




In [22]:
#tweets_http = df_tweets['tweets'].str.contains('https' ).sum()

In [23]:
#tweets_http

### Make a dictioary: Tokenize

In [24]:

list_words= ['turkey','islamicstate','syria','aleppo','warreporter1','russia', 'assad', 'iraq']
dict_token=defaultdict(list)
for value in list_words:
   for key in stopwords_morethan50:
      if value in key:           
           dict_token[value].append(key)
        
    
    

The words 'turkey','islamicstate','syria','aleppo','warreporter1','russia', 'assad', 'iraq' are found to be repeated in different formats

In [25]:
dict_token

defaultdict(list,
            {'turkey': ['turkey'],
             'islamicstate': ['islamicstate'],
             'syria': ['syria', 'syrian', 'syria:', 'syria.', 'syria|n'],
             'aleppo': ['aleppo', 'aleppo.'],
             'warreporter1': ['warreporter1:', 'warreporter1'],
             'russia': ['russia', 'russian', 'russia|n'],
             'assad': ['assad', 'assads'],
             'iraq': ['iraq', 'iraqi', 'iraq|i']})

In [30]:

from nltk.tokenize import TweetTokenizer
tweet_tokenizer = TweetTokenizer()
tweet_tokens = []
for elem in df_tweets['tweets']:
   # print(tweet_tokenizer.tokenize(elem))
    tweet_tokens.append(tweet_tokenizer.tokenize(elem))
tweet_tokens    



[['english',
  'translation',
  ':',
  'a',
  'message',
  'to',
  'the',
  'truthful',
  'in',
  'syria',
  '-',
  'sheikh',
  'abu',
  'muhammed',
  'al',
  'maqdisi',
  ':',
  'http://t.co/73xfszsjvr',
  'http://t.co/x8bzcscxzq'],
 ['english',
  'translation',
  ':',
  'sheikh',
  'fatih',
  'al',
  'jawlani',
  'for',
  'the',
  'people',
  'of',
  'integrity',
  ',',
  'sacrifice',
  'is',
  'easy',
  'http://t.co/uqqzxggvtz',
  'http://t.co/a7nbjwyhbr'],
 ['english',
  'translation',
  ':',
  'first',
  'audio',
  'meeting',
  'with',
  'sheikh',
  'fatih',
  'al',
  'jawlani',
  '(',
  'ha',
  '):',
  'http://t.co/tgxt1gdgw7',
  'http://t.co/zue8eisze6'],
 ['english',
  'translation',
  ':',
  'sheikh',
  'nasir',
  'al',
  'wuhayshi',
  '(',
  'ha',
  ')',
  ',',
  'leader',
  'of',
  'aqap',
  ':',
  'the',
  'promise',
  'of',
  'victory',
  ':',
  'http://t.co/3qg5dkliwr',
  'http://t.co/7bqk1wjazc'],
 ['english',
  'translation',
  ':',
  'aqap',
  ':',
  'response',
  'to'

In [31]:
tweet_tokens =[" ".join(x) for x in tweet_tokens]

#### Tf-idf

In [32]:
# Import TfidfVectorizer
from sklearn.feature_extraction.text import TfidfVectorizer 
stop_words_list = ['english', 'is', 'with', 'by', 'oh', 'have', 'been', 'will', 'if', 'will', 'if',
                   'else', 'so', 'we', 'then', 'let', 'us', 'and', 'that', 'or', 'do', 'has', 'on', 'from', 'those', 'are', 
                   'their', 'were', 'etc', 'now', 'just', 'as', 'here', 'upon', 'into', 'also', 'at', 'its', 'while',
                    'had', 'youre', 'la', 'be', 'can', 'did', 'too', 'your', 'which']

# Instantiate TfidfVectorizer
tv = TfidfVectorizer(max_features=1000, stop_words=stop_words_list) 

# Fit the vectroizer and transform the data
tv_transformed = tv.fit_transform(tweet_tokens )

# Create a DataFrame with these features
tv_df = pd.DataFrame(tv_transformed.toarray(), 
                     columns=tv.get_feature_names()).add_prefix('TFIDF_')
tv.vocabulary_


{'message': 558,
 'to': 880,
 'the': 863,
 'in': 432,
 'syria': 835,
 'sheikh': 778,
 'abu': 42,
 'al': 70,
 'http': 420,
 'co': 221,
 'for': 343,
 'people': 654,
 'of': 617,
 'first': 337,
 'leader': 504,
 'aqap': 107,
 'victory': 928,
 'statement': 815,
 'it': 457,
 'second': 760,
 'series': 767,
 'soldier': 794,
 'jn': 469,
 'video': 929,
 'link': 517,
 'words': 965,
 'dawlah': 247,
 'please': 669,
 'share': 773,
 'our': 639,
 'new': 599,
 'account': 45,
 'after': 53,
 'one': 629,
 'was': 943,
 '123': 7,
 'raid': 701,
 'france': 352,
 'taken': 839,
 'down': 279,
 'attack': 127,
 'mujahideen': 584,
 'https': 421,
 'saw': 753,
 'not': 609,
 'way': 945,
 'through': 876,
 'few': 328,
 'more': 574,
 'days': 249,
 'very': 925,
 'news': 600,
 'reports': 729,
 'syrian': 836,
 'should': 783,
 'still': 816,
 'path': 652,
 'watch': 944,
 'save': 752,
 'aleppo': 71,
 'released': 720,
 'group': 378,
 'islamic': 451,
 'no': 604,
 'rt': 738,
 'part': 650,
 '19': 14,
 'breaking': 177,
 'confirmed':

In [ ]:
# Isolate the row to be examined
sample_row = tv_df.iloc[0]

# Print the top 5 words of the sorted output
print(sample_row.sort_values(ascending=False).head(5))


In [ ]:
import